### PR TITLE
[XLink] - Fix Mac build

### PR DESCRIPTION
**Pull request by DariaMityagina**

### CODE REVIEW

File: inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c [hunk 1]
```diff
@@ -24,6 +24,7 @@
 # include "win_semaphore.h"
 #else
 # include <pthread.h>
+# include <unistd.h>
```
Comment (AntonDudchenko) [inline inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c:27]: Shouldn't we put this under ifdef __APPLE__, or it's required for Linux as well?

Comment (DariaMityagina) [reply inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c:27]: It won't hurt Linux:)

**Review by AntonDudchenko — APPROVED** [review]

File: inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c [hunk 2]
```diff
@@ -398,7 +399,11 @@ int DispatcherWaitEventComplete(xLinkDeviceHandle_t *deviceHandle, unsigned int
         // This is a workaround for sem_timedwait being influenced by the system clock change.
         // This is a temporary solution. TODO: replace this with something more efficient.
         while (timeoutMs--) {
+#if (defined(__APPLE__))
+            rc = sem_wait(id->psem);
```
Comment (alalek) [inline inference-engine/thirdparty/movidius/XLink/shared/src/XLinkDispatcher.c:403]: timeout handling?

**Comment by DariaMityagina** [comment]
Accidently closed this. You can find corresponding changes here: https://github.com/openvinotoolkit/openvino/pull/8479

**Pull request closed without merging**